platforms/documentation/docs/src/docs/userguide/dep-man/understanding-dependency-resolution/dependency_caching.adoc

```diff
@@ -0,0 +1,265 @@
+
+This concurrent access is only supported if the different Gradle processes can communicate together. This is usually _not the case_ for containerized builds.
+
+[[sub:cache_cleanup]]
+=== Cache Cleanup
+
+Gradle tracks which artifacts in the dependency cache are accessed. Based on this information, the cache is periodically scanned (no more than once every 24 hours) to identify artifacts that haven't been used in over 30 days.
+These obsolete artifacts are then deleted to prevent the cache from growing indefinitely.
+
+[[sec:controlling-dynamic-version-caching]]
+== 2. Controlling the cache
+
+By default, Gradle caches dependencies (including dynamic versions and changing modules) for 24 hours, meaning it does not contact remote repositories for new versions during this time.
+
+To have Gradle check for newer versions more frequently or with every build, you can adjust the caching threshold or time-to-live (TTL) settings accordingly.
+
+NOTE: Using a short TTL threshold for dynamic or changing versions may result in longer build times due to increased HTTP(s) calls.
+
+[[sec:controlling_dependency_caching_programmatically]]
+=== Controlling dependency caching programmatically
+
+You can fine-tune certain aspects of caching programmatically using the link:{javadocPath}/org/gradle/api/artifacts/ResolutionStrategy.html[ResolutionStrategy] for a configuration.
+The programmatic approach is useful if you want to change the settings permanently.
+
+To change how long Gradle will cache the resolved version for a dynamic version, use:
+
+====
+include::sample[dir="snippets/dependencyManagement/troubleshooting-cache-dynamic/kotlin",files="build.gradle.kts[tags=dynamic-version-cache-control]"]
+include::sample[dir="snippets/dependencyManagement/troubleshooting-cache-dynamic/groovy",files="build.gradle[tags=dynamic-version-cache-control]"]
+====
+
+To change how long Gradle will cache the metadata and artifacts for a changing module, use:
+
+====
+include::sample[dir="snippets/dependencyManagement/troubleshooting-cache-changing/kotlin",files="build.gradle.kts[tags=changing-module-cache-control]"]
+include::sample[dir="snippets/dependencyManagement/troubleshooting-cache-changing/groovy",files="build.gradle[tags=changing-module-cache-control]"]
+====
+
+== 3. Caching changing dependencies
+
+Gradle treats dependencies marked as "changing" (such as SNAPSHOT dependencies) differently from regular dependencies, refreshing them more frequently to ensure that you are always using the latest version.
+
+To declare a dependency as changing, you can set the `changing = true` attribute in your dependency declaration.
+This is useful for dependencies expected to change frequently without a new version number:
+
+[source,groovy]
+----
+dependencies {
+    implementation("com.example:my-library:1.0-SNAPSHOT") {
+        changing = true
+    }
```

Comment:
This is a bad example, as we already treat these as `changing`. There's no need to declare this explicitly.

Reply:
Updated the example.